server/middleware/journey/journeyStateGuard.ts

```diff
@@ -0,0 +1,56 @@
+import type { NextFunction, Request, Response } from 'express'
+import { validate } from 'uuid'
+
+export type JourneyStateGuard = { [pageName: string]: (req: Request) => string | undefined }
+
+export function isMissingValues<T>(obj: T, keys: Array<keyof T>): boolean {
+  return keys.some(key => obj?.[key] === undefined)
+}
+
+export default function journeyStateGuard(rules: JourneyStateGuard) {
+  return (req: Request, res: Response, next: NextFunction): void => {
+    const uuidMatch = req.originalUrl.match(/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/)
+    const uuid = uuidMatch?.[0]
+
+    if (!uuid || !validate(uuid)) {
+      // This page does not concern us
+      return next()
+    }
+
+    const requestedPage = req.originalUrl.split('/').pop()
+    const flow = req.originalUrl.substring(0, uuidMatch.index - 1)
+
+    if (!requestedPage || !flow) {
```

Comment:
You've added some useful comments here about what the sections are doing, but I think it could do with some more? Just a line to indicate what circumstance each bit is catching please?

eg. It's not immediately what sections like this are doing..
`if (!requestedPage || !flow) {
  return next()
}`